src/ralph/cli.py

```diff
@@ -310,11 +309,14 @@ def fetch(backend, archive, chunk_size, **options):
     backend_type = get_backend_type(backend_class)
 
     if backend_type == BackendTypes.STORAGE:
-        backend.read(archive, chunk_size=chunk_size)
+        for data in backend.read(archive, chunk_size=chunk_size):
+            click.echo(data, nl=False)
     elif backend_type == BackendTypes.DATABASE:
-        backend.get(chunk_size=chunk_size)
+        for document in backend.get(chunk_size):
+            source = json.dumps(document.get("_source"))
```

Comment:
This will only work for the ES backend right?

Reply:
:thinking: After thought: I would prefer streaming hits as is and use a dedicated parser with the `extract` command instead. I have the feeling that we will end up with unmaintainable spaghetti code if we use this pattern. WDYT?

Reply:
Do we replace:
```python
source = json.dumps(document.get("_source"))
```
With:
```python
source = json.dumps(document)
```
And create an `ElasticSearchParser` class to do the `.get("_source")` part ?
Or do we change the `ESDatabase.get` method to return strings instead of objects?
Or maybe we could add a switch argument like `to_string=True` in the `ESDatabase.get` method to benefit both formats (string and object)?
Sorry, I have more questions than answers :sweat_smile:

Reply:
IMO a backend's `get` method should return raw records, and we should use parser to get core (xAPI) events.

Reply:
Ok, updated this part to make the `fetch` command return the raw ES records and added an `ElasticsearchParser` to extract the `_source` field.